### DISCUSSION TITLE
Snapshot release question.

### DISCUSSION THREAD

**Discussion by guangster** [bytedeco/javacpp-presets#1131] [Q&A] [answered]
Hi @saudet, will there be a release soon for 1.5.7 SNAPSHOT?

I'm running into some issues as I think the GPU version and the regular PyTorch versions are different if I check http://bytedeco.org/builds/. I think GPU is using torch 1.9 while pytorch-platform is on 1.10?
I'm also running things on different machines and I noticed the newer JavaCPP SNAPSHOTS have started to use newer PyTorch versions.

**Comment by saudet** [comment]
There are snapshots for 1.10.1: https://oss.sonatype.org/content/repositories/snapshots/org/bytedeco/pytorch/1.10.1-1.5.7-SNAPSHOT/

**Comment by saudet** [comment]
If you're asking about the "-gpu" artifacts, those take more than 6 hours to build on GitHub Actions, so not available as snapshots.
Please use binaries from [LibTorch](https://pytorch.org/cppdocs/installing.html) for this purpose: https://github.com/bytedeco/javacpp-presets/tree/master/pytorch#documentation

**Reply by guangster** [reply]
Thanks for your quick response as always.

So I was trying some code on different machines and wanted to run the 1.5.7-SNAPSHOT version of JavaCPP Pytorch Preset because it had some changes I needed.

However, the latest snapshots on Sonatype update the Pytorch version which makes things a bit difficult. I was wondering if there would be a formal release and I can just get it from the maven central repo with a static Pytorch version, or if there's a way I can keep using a fixed Pytorch setup to work with JavaCPP?

Thanks!

**Reply by saudet** [reply]
You can backport changes to a previous version, if that's what you mean, yes. I believe that's what @carlosuc3m is attempting to do.

**Reply by carlosuc3m** [reply]
That's it. I've been able to create the presets for Pytorch>=1.8.0 with Javacpp-1.5.7-SNAPSHOT following the documentation:
https://github.com/bytedeco/javacpp-presets/wiki/Create-New-Presets
The trickiest part for me is updating the Java configuration file. However for  versions 1.8 and 1.9 I have just used the ones that had already be done for other Javacpp versions:
https://github.com/bytedeco/javacpp-presets/blob/ded739e5d93525ec5e3c3a84b236df7fc121b9e8/pytorch/src/main/java/org/bytedeco/pytorch/presets/torch.java for 1.9.0

https://github.com/bytedeco/javacpp-presets/blob/41f5aee18f2f95be2b14a4e8609fda76c9cebf61/pytorch/src/main/java/org/bytedeco/pytorch/presets/torch.java for 1.8.1 

REgards,
Carlos

**Comment by saudet** [comment, marked as answer]
I have been able to get CUDA-enabled snapshots of the presets for PyTorch along with the rest as well deployed automatically using GitHub Actions, see commit 3db41e82c093299d88ecf77c9f3d2622fd962219, so they should now always be up-to-date. Enjoy!
